easybuild/easyconfigs/s/SciPy-bundle/SciPy-bundle-2023.12-iimkl-2023b.eb

```diff
@@ -0,0 +1,117 @@
+easyblock = 'PythonBundle'
+
+name = 'SciPy-bundle'
+version = '2023.12'
+
+homepage = 'https://python.org/'
+description = "Bundle of Python packages for scientific software"
+
+toolchain = {'name': 'iimkl', 'version': '2023b'}
+toolchainopts = {
+    'pic': True, 'lowopt': True, 'strict': True,
+    'oneapi': True, 'optarch': False,  # prevents -xHost
+}
+
+builddependencies = [
+    ('hypothesis', '6.90.0'),
+    ('UnZip', '6.0'),
+    # scipy >= 1.9.0 uses Meson/Ninja
+    ('Meson', '1.2.3'),
+    ('meson-python', '0.15.0'),
+    ('Ninja', '1.11.1'),
+    ('pkgconf', '2.0.3'),   # required by scipy
+]
+
+dependencies = [
+    ('Python', '3.11.5'),
+    ('Python-bundle-PyPI', '2023.10'),
+    ('pybind11', '2.11.1'),  # required by scipy
+]
+
+use_pip = True
+
+# order is important!
+exts_list = [
+    ('numpy', '1.26.2', {
+        'easyblock': 'PythonPackage',  # pip install builds numpy v1.26.x via spin/meson/ninja
```

Comment:
[This feature isn't really documented](https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyblock.py#L593)

Am I allowed to use it?

Reply:
It's not disallowed, though this would mean we e.g. aren't running test step (and all the other good stuff that presumably is done in that easyblock, like picking up fft etc.)

I would like to input from one of the authors of the numpy easyblock. I only recognize @boegel names there.

Reply:
Specifying a custom easyblock for a particular extension like this is fine in general, but bypassing the custom easyblock for `numpy` specifically is a bit of a "red flag" (strongly worded).

It seems like that may point to a fix being needed in the numpy easyblock instead?

Reply:
> It seems like that may point to a fix being needed in the numpy easyblock instead?

Yes, that's what I thought first too. One might say that the `numpy` easyblock need to evolve along with the [changes in `numpy`'s build system (v1.26.0)](https://numpy.org/doc/2.1/release/1.26.0-notes.html#build-system-changes).

Similarly to [what happened to the `SciPy` easyblock (pr 2862)](easybuilders/easybuild-easyblocks/pull/2862), when there were [changes in `SciPy`'s build system (v1.9)](https://scipy.github.io/devdocs/release/1.9.0-notes.html#scipy-switched-to-meson-as-its-build-system).

But I'm in doubt, because the new build system is (advertised as) simpler, then why would we making our easyblock more complex?  If the generic `pythonpackage` works better than the bespoke `numpy` block, we have an opportunity to be leaner. 

I want to do what happened with SciPy block, but I want it to be a oneliner. One test that makes it revert to the generic case, say `if LooseVersion(self.version) >= LooseVersion('1.26')`. I don't like the idea of adding many `if self.use_meson` statements in the block, doubling it, when above I 

Am I missing something? Those tests that @Micket mentioned? Please let me know what you think.